website/src/codegen/dpp/main.cpp.ejs

```diff
@@ -0,0 +1,21 @@
+#include <dpp/dpp.h>
+    
+int main() {
+    dpp::cluster bot("token");
+
+    const dpp::snowflake SOME_CHANNEL_ID(123456789012345);
+    
+    bot.on_log(dpp::utility::cout_logger());
+    
+    bot.on_ready([&bot](const auto& event) {
+        auto m = dpp::message()
+            .set_flags(dpp::m_using_components_v2)
+            // This method allows you to add arbitrary JSON for components.
+            // It is ideal for pre-generated content such as this. If you need to change
+            // components on the fly, please see https://dpp.dev/classdpp_1_1component.html
+            .add_json_components("<%- JSON.stringify(data.components).replace(/[\n"\&\r\t\b\f]/g, '\\$&') %>");
+        bot.message_create(SOME_CHANNEL_ID, m);
+    });
+    
+    bot.start(dpp::st_wait);
+}
```

Comment:
```suggestion
auto m = dpp::message()
    .set_flags(dpp::m_using_components_v2)
    // This method allows you to add arbitrary JSON for components.
    // It is ideal for pre-generated content such as this. If you need to change
    // components on the fly, please see https://dpp.dev/classdpp_1_1component.html
    .add_json_components("<%- JSON.stringify(data.components).replace(/[\n"\&\r\t\b\f]/g, '\\$&') %>");
```

My two cents, but ultimately not my choice. I like more being able to just press the "copy button" (pr open for it) and paste it directly where it needs to go in the code.

It also makes it explicitly clear what this is for, and not a full blown bot generator

Reply:
if thats the case why do the hikari ones have an import? :)
I think its important to make context clear for dpp users, or we will get people just making a .cpp with literally that one line in it...

Reply:
> if thats the case why do the hikari ones have an import? :)

That was a mistake, updated! Import for sure needs to be there :P

> I think its important to make context clear for dpp users, or we will get people just making a .cpp with literally that one line in it...

It might be confusing for some, but if we set the president for all of the outputs to be strictly just the components list/builder that later gets passed to the appropriate call, should confuse users way less than "this is a json and then when you press a tab you get a fully functional bot to test this".

It will also make it nicer to iterate over components and copy them directly into the project